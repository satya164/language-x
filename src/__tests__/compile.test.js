/* @flow */

const dedent = require('dedent');
const parse = require('../parse');
const compile = require('../compile');

it('compiles program to JavaScript', () => {
  expect(
    compile(
      parse(dedent`
    let a = 10 * 30 + b - 20 / foo
    let b = "Hello world"

    fun add a b = {
      let c = a + b

      return c
    }

    fun foo = add 3 4

    main add 3 4
  `)
    )
  ).toMatchSnapshot();
});
