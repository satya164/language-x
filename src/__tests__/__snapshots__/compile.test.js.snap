// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiles program to JavaScript 1`] = `
"'use strict';

var a = 10 * 30 + b - 20 / foo;

var b = \\"Hello world\\";

function add(a, b) {
  var c = a + b;

  return c;
}

function foo() { return add(3, 4); }

add(3, 4);"
`;
